### ISSUE TITLE
Build fails with "undefined reference to `__stack_chk_fail'"

### ISSUE THREAD

**Issue by fhs** [Harvey-OS/harvey#850]
#### Expected behavior

Successful build

#### Actual behavior
```
2018/09/25 18:25:14 Linking Cmd
2018/09/25 18:25:14 /usr/bin/ld -o basename basename.o -L /home/big/src/harvey/amd64/lib -static -e_main -lavl -lcomplete -lcontrol -ljay -ldisk -lflate -lframe -lfreetype -lgeometry -lhttpd -lbin -llex -lmach -lmemdraw -lmemlayer -lmemdraw -ldraw -lplumb -lregexp -lString -loventi -l9p -lauth -lauthsrv -lndb -lip -lventi -lsec -lstdio -lauth -lmp -lthread -lbio -lc
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(print.o): in function `print':
/home/big/src/harvey/sys/src/libc/fmt/print.c:23: undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(utfrrune.o): in function `utfrrune':
/home/big/src/harvey/sys/src/libc/port/utfrrune.c:39: undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(fprint.o): in function `fprint':
/home/big/src/harvey/sys/src/libc/fmt/fprint.c:23: undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(rune.o): in function `chartorune':
/home/big/src/harvey/sys/src/libc/port/rune.c:82: undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(rune.o): in function `runelen':
/home/big/src/harvey/sys/src/libc/port/rune.c:134: undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libc.a(vfprint.o):/home/big/src/harvey/sys/src/libc/fmt/vfprint.c:45: more undefined references to `__stack_chk_fail' follow
2018/09/25 18:25:14 exit status 1
```
#### Steps to reproduce
- ./bootstrap.sh
- CC=gcc ARCH=amd64 util/build

Same error when I tried clang.
```
$ gcc --version
gcc (GCC) 8.2.1 20180831
...
$ clang --version
clang version 6.0.1 (tags/RELEASE_601/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by elbing** [comment]
Please, test with these PR: https://github.com/Harvey-OS/harvey/pull/851
And tell us here.

**Comment by fhs** [comment]
It got further with the PR. It fails with this now:
```
2018/09/26 12:00:46 Building irq
2018/09/26 12:00:46 /usr/bin/gcc -std=c11 -c -I /home/big/src/harvey/amd64/include -I /home/big/src/harvey/sys/include -I . -DACPI_DEBUGGER -DACPI_DISASSEMBLER -DACPI_EXEC_APP -D__HARVEY__ -U_LINUX -U__linux__ -U__FreeBSD__ -U__NetBSD__ -Wno-unused-function -Wno-unused-variable -Wno-unused-const-variable -Wno-unknown-pragmas -Wall -Werror -nostdlib -nostdinc -fno-builtin -include u.h -include libc.h -include ctype.h -include /home/big/src/harvey/amd64/include/mach_acpi.h -I /home/big/src/harvey/sys/include/acpi/acpica -ffreestanding -fno-builtin -fno-omit-frame-pointer -fno-stack-protector -g -gdwarf-2 -ggdb -mcmodel=small -mno-red-zone -O0 -static -Wall -Werror -fno-pie -fvar-tracking -fvar-tracking-assignments TestACPISleep.c
2018/09/26 12:00:46 /usr/bin/gcc -std=c11 -c -I /home/big/src/harvey/amd64/include -I /home/big/src/harvey/sys/include -I . -DACPI_DEBUGGER -DACPI_DISASSEMBLER -DACPI_EXEC_APP -D__HARVEY__ -U_LINUX -U__linux__ -U__FreeBSD__ -U__NetBSD__ -Wno-unused-function -Wno-unused-variable -Wno-unused-const-variable -Wno-unknown-pragmas -Wall -Werror -nostdlib -nostdinc -fno-builtin -include u.h -include libc.h -include ctype.h -include /home/big/src/harvey/amd64/include/mach_acpi.h -I /home/big/src/harvey/sys/include/acpi/acpica -ffreestanding -fno-builtin -fno-omit-frame-pointer -fno-stack-protector -g -gdwarf-2 -ggdb -mcmodel=small -mno-red-zone -O0 -static -Wall -Werror -fno-pie -fvar-tracking -fvar-tracking-assignments crs.c
2018/09/26 12:00:46 /usr/bin/gcc -std=c11 -c -I /home/big/src/harvey/amd64/include -I /home/big/src/harvey/sys/include -I . -DACPI_DEBUGGER -DACPI_DISASSEMBLER -DACPI_EXEC_APP -D__HARVEY__ -U_LINUX -U__linux__ -U__FreeBSD__ -U__NetBSD__ -Wno-unused-function -Wno-unused-variable -Wno-unused-const-variable -Wno-unknown-pragmas -Wall -Werror -nostdlib -nostdinc -fno-builtin -include u.h -include libc.h -include ctype.h -include /home/big/src/harvey/amd64/include/mach_acpi.h -I /home/big/src/harvey/sys/include/acpi/acpica -ffreestanding -fno-builtin -fno-omit-frame-pointer -fno-stack-protector -g -gdwarf-2 -ggdb -mcmodel=small -mno-red-zone -O0 -static -Wall -Werror -fno-pie -fvar-tracking -fvar-tracking-assignments irq.c
2018/09/26 12:00:46 Linking irq
2018/09/26 12:00:46 /usr/bin/ld -o TestACPISleep TestACPISleep.o -L /home/big/src/harvey/amd64/lib -lacpi -static -e_main -lavl -lcomplete -lcontrol -ljay -ldisk -lflate -lframe -lfreetype -lgeometry -lhttpd -lbin -llex -lmach -lmemdraw -lmemlayer -lmemdraw -ldraw -lplumb -lregexp -lString -loventi -l9p -lauth -lauthsrv -lndb -lip -lventi -lsec -lstdio -lauth -lmp -lthread -lbio -lc
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(tbxfload.o): in function `AcpiInstallTable':
tbxfload.c:(.text+0x51e): undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(tbxfload.o): in function `AcpiLoadTable':
tbxfload.c:(.text+0x6cd): undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(evregion.o): in function `AcpiEvAddressSpaceDispatch':
evregion.c:(.text+0x58b): undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(evregion.o): in function `AcpiEvExecuteRegMethod':
evregion.c:(.text+0xdaa): undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(evregion.o): in function `AcpiEvExecuteRegMethods':
evregion.c:(.text+0xf0b): undefined reference to `__stack_chk_fail'
/usr/bin/ld: /home/big/src/harvey/amd64/lib/libacpi.a(evregion.o):evregion.c:(.text+0x114d): more undefined references to `__stack_chk_fail' follow
2018/09/26 12:00:46 exit status 1
```

**Comment by elbing** [comment]
Oh well, ACPI uses its own flags json. Updated :-) , try again, please.

**Comment by elbing** [comment]
Added! thanks for the tip. Enjoy it :-)

**Comment by fhs** [comment]
Thanks. Closing since the fix (#851) has been merged.
